### DISCUSSION TITLE
How does it work?

### DISCUSSION THREAD

**Discussion by Zorral** [Vedza/NitroSniperGo#227] [General]
After starting up the sniper, how does it start to work? Like do I have to manually join servers and enter nitro gaws or does it join automatically?

**Comment by Vedza** [comment]
It does everything automatically

**Reply by Zorral** [reply]
Im on Mac and it automatically ran the code through Terminal, but it doesn't seem to join servers on its own. Do I have to add any additional settings or set up an auto server joiner and gaw joiner?

**Reply by Vedza** [reply]
Just read the settings here https://github.com/Vedza/NitroSniperGo#usage
There is a delay to join servers and invite sniper must be enabled.

**Reply by Zorral** [reply]
alright thanks

**Reply by Zorral** [reply]
11:27:19 [x] Failed to parse JSON file:  invalid character 'a' looking for beginning of object key string
The showed up after I set up everything.

**Reply by Zorral** [reply]
that issue somehow fixed itself, but how do you know if it's working?

**Reply by Vedza** [reply]
If you see `[+] Sniper is ready` then it's working

**Reply by Zorral** [reply]
2021/03/29 15:54:46 [DG0] wsapi.go:574:onEvent() error unmarshalling READY event, json: cannot unmarshal number into Go struct field ReadState.read_state.last_message_id of type string
this showed up under [+] Sniper is ready.

**Reply by Zorral** [reply]
> If you see `[+] Sniper is ready` then it's working

Will it automatically start joining gaws?

**Reply by Zorral** [reply]
2021/03/29 15:54:41 [DG0] wsapi.go:574:onEvent() error unmarshalling READY event, json: cannot unmarshal number into Go struct field ReadState.read_state.last_message_id of type string
Sniping Discord Nitro and Giveaway for **** on 13 servers and 2 accounts 🔫

15:54:44 [+] Sniper is ready
2021/03/29 15:54:46 [DG0] wsapi.go:574:onEvent() error unmarshalling READY event, json: cannot unmarshal number into Go struct field ReadState.read_state.last_message_id of type string

But this last part showed up so I don't think it's working. Any fixes?

**Reply by Vedza** [reply]
https://github.com/Vedza/NitroSniperGo#known-issues

**Reply by Zorral** [reply]
> https://github.com/Vedza/NitroSniperGo#known-issues

im so F***** stupid... I was just reading it. Sorry for all these questions and the inconvenience, the code seems to be working out great! (thankkkkksss!) But I don't know if it's auto joining gaws since the only things that have showed up is this.  

17:38:37 [-] **** sniped code: UEBTQbruYXG6PJkq from Kamaboko Gonpachiro#1407 [Bedless Nation > 💬english-chat]
17:38:37 [-] This gift has been redeemed already. Delay: 268ms
17:38:37 [-]**** sniped code: UEBTQbruYXG6PJkq from Kamaboko Gonpachiro#1407 [Bedless Nation > 💬english-chat]
17:38:37 [-] This gift has been redeemed already. Delay: 319ms
